src/helpers/createNavigationHelpers.ts

```diff
@@ -0,0 +1,185 @@
+  pop: (mergeOptions?: Options) => Promise<any>
+  popTo: (mergeOptions?: Options) => Promise<any>
+  popToRoot: (mergeOptions?: Options) => Promise<any>
+  dismissOverlay: () => Promise<any>
+  dismissAllModals: (mergeOptions?: Options) => Promise<any>
+  getLaunchArgs: () => Promise<any>
+  events: () => EventsRegistry
+}
+
+/**
+ * Creates a set of action helpers for
+ * [component](https://wix.github.io/react-native-navigation/api/component),
+ * [root](https://wix.github.io/react-native-navigation/api/root),
+ * [stack](https://wix.github.io/react-native-navigation/api/stack),
+ * [modal](https://wix.github.io/react-native-navigation/api/modal),
+ * [overlay](https://wix.github.io/react-native-navigation/api/overlay)
+ * Navigation API
+ */
+function createRnnHelpers(
+  /**
+   * Component reference id. Used to give context to the Navigation functions that requires componentId parameter.
+   */
+  componentId: string
+): NavigationHelpers {
+  function setRoot<P = {}>(nameOrLayout: string | Layout | LayoutRoot, passProps?: P, options?: Options) {
+    let layoutRoot
+
+    if (typeof nameOrLayout === 'string') {
+      layoutRoot = { root: createLayout<P>(nameOrLayout, passProps, options) } as LayoutRoot
+    } else if ((nameOrLayout as LayoutRoot).root) {
+      layoutRoot = nameOrLayout as LayoutRoot
+    } else {
+      layoutRoot = { root: nameOrLayout } as LayoutRoot
+    }
+
+    return Navigation.setRoot(layoutRoot)
+  }
+
+  function setStackRoot<P = {}>(nameOrLayout: string | Layout<P> | Array<Layout<P>>, passProps?: P, options?: Options) {
+    const layout = typeof nameOrLayout === 'string' ? createLayout<P>(nameOrLayout, passProps, options) : nameOrLayout
+
+    return Navigation.setStackRoot(componentId, layout)
+  }
+
+  function push<P = {}>(nameOrLayout: string | Layout<P>, passProps?: P, options?: Options) {
+    const layout = typeof nameOrLayout === 'string' ? createLayout<P>(nameOrLayout, passProps, options) : nameOrLayout
+
+    return Navigation.push(componentId, layout)
+  }
+
+  function showModal<P = {}>(nameOrLayout: string | Layout<P>, passProps?: P, options?: Options) {
+    const layout = typeof nameOrLayout === 'string' ? createLayout<P>(nameOrLayout, passProps, options) : nameOrLayout
+
+    return Navigation.showModal(layout)
+  }
+
+  function showModalStack<P = {}>(
+    nameOrLayoutStackChildren: string | LayoutStackChildren | LayoutStackChildren[],
+    passProps?: P,
+    options?: Options
+  ) {
+    let layoutStackChildren
+
+    if (typeof nameOrLayoutStackChildren === 'string') {
+      const layoutComponent = createLayout<P>(nameOrLayoutStackChildren, passProps, options) as LayoutStackChildren
+      layoutStackChildren = [layoutComponent]
+    } else {
+      layoutStackChildren = Array.isArray(nameOrLayoutStackChildren)
+        ? nameOrLayoutStackChildren
+        : [nameOrLayoutStackChildren]
+    }
+
+    return Navigation.showModal({ stack: { children: layoutStackChildren } })
```

Comment:
What we're missing here is setting some variables that are _siblings_ of `stack`.
I guess it could be a trade-off when using these helpers, but It'd be good to think if we could add support for that case without adding complexity to the API.

Reply:
This is intended to be a shortcut to reduce the boilerplate for common cases of `showModal`. If you wanna provide stack siblings you could always use `showModal` instead. 

BTW I will think if I can improve the interface for this case

Reply:
What I'm thinking is about the other options you have when calling `showModal`.
```ts
Navigation.showModal({
  stack: { 
    id: 'some-id',
    options: {
      // ... styling/behavior options
    }
  }
})
```
In that case, this helper wouldn't actually help 😄 

So, I'm wondering if `showModalStack` shouldn't be part of the project using this library instead.
Otherwise we'd need to support this... maybe providing a recipe would be better.

Reply:
In that case you could use: 

```ts
const { showModal } = useNavigation()

showModal('ScreenName', { prop1: 'prop-value-1' }, { option1: 'option-value-1' })
```

makes sense?

Reply:
Those options are not applied to the stack but to the specific component.
Also, there is an `id` field alongside the `options`.

Consider this scenario
```js
Navigation.showModal({
  stack: { 
    id: 'some-id', // can't set this
    options: {...}, // can't set this
    children: [
      ...
    ]
  }
})
```

Reply:
In this case I would rather use the out-of-the-box functionality isn't it?

```tsx
interface ShowModalCommand {
  (layout: Layout): Promise<any> // <<< This overload
  <P = {}>(name: string, passProps?: P, options?: Options): Promise<any>
}
```


![image](https://user-images.githubusercontent.com/95109/90046070-4e9fd180-dca6-11ea-8e25-914f2e2f0c81.png)

Reply:
Totally! 👍 
I think the only thing I've with `showModalStack` is that it's opinionated while the other methods aren't, and that could be confusing 🤔 I mean, it's not exposing a RNN method directly, but a convention. And hundreds of these conventions could exists, so I don't see where is the line.

Reply:
The following PR includes the changes we were discussing 2b6ea18aa93ebf81862b983ee00fd4fc30416169

Thanks for the feedback!